### PR TITLE
chore: swagger docs omit brower based credentials, rely on swagger auth

### LINKED ISSUE #13535: Swagger CSRF Error
Some requests don't work due missing of the X-CSRF-TOKEN header. 

E. g. `.../swagger/index.html#/Builds/create-workspace-build`.

The 400 response contains the message 

`Something is wrong with your CSRF token. Please refresh the page. If this error persists, try clearing your cookies.`

Neither a page refresh nor clearing the cookies solves the problem.

An curl command of the exact same request, but including the X-CSRF-TOKEN header with the content of the csrf_token cookie works as expected.

### PR DESCRIPTION
**Pull request by Emyrk**
Swagger has an "Authorize" button which should be the only authentication being used in the api requests

Closes https://github.com/coder/coder/issues/13535

Note: I considered implementing CSRF in the interceptor which would just use the logged in user credentials. But the `Authorize` button will still exist, and it would be even more confusing since the cookie auth supersedes the header based auth. So swagger requiring explicit authentication feels safer.

### CODE REVIEW

File: coderd/coderd.go [hunk 1]
```diff
@@ -87,7 +87,31 @@ import (
...
+		// CSRF checks **if** there is no cookie auth also present.
+		// (If the cookie matches, then it's ok too)
+		//
+		// Because swagger is hosted on the same domain, we have the cookie
+		// auth and the header auth competing. This can cause CSRF errors,
+		// and can be confusing what authentication is being used.
+		//
+		// So remove authenticating via a cookie, and rely on the authorization
+		// header passed in.
+		httpSwagger.UIConfig(map[string]string{
```
Comment (aslilac) [inline coderd/coderd.go:103]: I hate that they use a `map[string]string` for this instead of a struct 😭

File: coderd/coderd.go [hunk 2]
```diff
@@ -87,7 +87,31 @@ import (
...
+			// Pulled from https://swagger.io/docs/open-source-tools/swagger-ui/usage/configuration/
+			// 'withCredentials' should disable fetch sending browser credentials, but
+			// for whatever reason it does not.
+			// So this `requestInterceptor` ensures browser credentials are
+			// omitted from all requests.
+			"requestInterceptor": `(a => {
+				a.credentials = "omit";
+				return a;
+			})`,
```
Comment (aslilac) [inline coderd/coderd.go:112]: they're recommendation is just to embed javascript in a string??? I extra hate that

**Review by aslilac — APPROVED** [review]
this is awful, but it seems like it's all in ways that aren't your fault, so whatever 😂

File: coderd/coderd.go (hunk 2 above)
Comment (Emyrk) [reply coderd/coderd.go:112]: Yea, it is a bit crazy: https://github.com/swaggo/http-swagger/blob/master/swagger.go#L24-L27

**Pull request merged**